modules/services/workspace-service/src/main/java/org/ballerinalang/composer/service/workspace/rest/datamodel/BLangFileRestService.java

```diff
@@ -145,13 +171,125 @@ public Response optionsParseFragment() {
      * @throws IOException
      */
     private static String parseJsonDataModel(InputStream stream, java.nio.file.Path filePath) throws IOException {
+        String tmpDir = System.getProperty("java.io.tmpdir");
+        CompilerContext context = new CompilerContext();
+        CompilerOptions options = CompilerOptions.getInstance(context);
+        File tmpFile = File.createTempFile("composer.tmp.", ".bal", new File(tmpDir));
+        OutputStream outputStream = new FileOutputStream(tmpFile);
+        IOUtils.copy(stream, outputStream);
+        outputStream.close();
```

Comment:
You can use https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/IOUtils.html#closeQuietly(java.io.Closeable...) here so that it wont throw the IOException.

Only if it is needed.